src/parser.rs

```diff
@@ -383,11 +383,10 @@ asdf(", "ASDF",
             |store: &mut Store<Fr>, input: &str, expected_ptr: Ptr<Fr>, expected_meta: bool| {
                 let mut chars = input.chars().peekable();
 
-                match store.read_maybe_meta(&mut chars).unwrap() {
-                    (ptr, meta) => {
-                        assert_eq!(expected_ptr, ptr);
-                        assert_eq!(expected_meta, meta);
-                    }
+                let (ptr, meta) = store.read_maybe_meta(&mut chars).unwrap();
+                {
+                    assert_eq!(expected_ptr, ptr);
+                    assert_eq!(expected_meta, meta);
                 };
```

Comment:
Out of curiosity, did Clippy make you change this?

Reply:
Yes.

Reply:
Clippy hates lexical scope, I guess.